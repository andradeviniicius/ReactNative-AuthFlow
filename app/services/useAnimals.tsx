import { useQuery } from "@tanstack/react-query";
import axios from "axios";
import { API_URL } from "../context/AuthContext";

export interface Animal {
  age: number;
  categoryId: number;
  id: string;
  img: string;
  name: string;
}
const useAnimals = () => {
  const getAnimals = () => {
    const result = axios.get<Animal[]>(API_URL + "/animals/");

    return result;
  };

  const { data, isError, isLoading } = useQuery({
    queryKey: ["animals"],
    queryFn: getAnimals,
  });

  return {
    animals: data?.data,
    isAnimalsError: isError,
    isAnimalsLoading: isLoading,
  };
};

export default useAnimals;
