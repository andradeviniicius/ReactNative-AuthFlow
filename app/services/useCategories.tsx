import { useQuery } from "@tanstack/react-query";
import axios from "axios";
import { API_URL } from "../context/AuthContext";

const useCategories = () => {
  const getCategories = () => {
    const result = axios.get(API_URL + "/categories");

    return result;
  };

  const { data, isError, isLoading } = useQuery({
    queryKey: ["categories"],
    queryFn: getCategories,
  });

  return { data, isError, isLoading };
};

export default useCategories;
