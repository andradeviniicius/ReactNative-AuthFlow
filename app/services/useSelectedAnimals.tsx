import { useQuery } from "@tanstack/react-query";
import axios from "axios";
import { API_URL } from "../context/AuthContext";

export interface Animal {
  age: number;
  categoryId: number;
  id: string;
  img: string;
  name: string;
}
export interface AnimalDetails extends Animal {
  description?: string;
  phone?: string;
  email?: string;
}
interface UseAnimalsProps {
  animalId: string;
}
const useSelectedAnimals = (props: UseAnimalsProps) => {
  const getAnimals = () => {
    const result = axios.get<AnimalDetails>(
      API_URL + "/animals/" + props.animalId
    );

    return result;
  };

  const { data, isError, isLoading } = useQuery({
    queryKey: ["selectedAnimals"],
    queryFn: getAnimals,
  });

  return {
    selectedAnimalDetails: data?.data,
    isSelectedAnimalsError: isError,
    isSelectedAnimalsLoading: isLoading,
  };
};

useSelectedAnimals.defaultProps = {
  animalId: "",
};

export default useSelectedAnimals;
