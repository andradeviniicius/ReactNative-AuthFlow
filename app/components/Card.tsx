import {
  View,
  Text,
  Image,
  StyleSheet,
  TouchableWithoutFeedback,
} from "react-native";
import React from "react";
import { AnimalDetails } from "../services/useAnimals";
import { useNavigation } from "@react-navigation/native";

interface CardProps {
  animal: AnimalDetails;
}

const Card = ({ animal }: CardProps) => {
  const navigation = useNavigation();

  return (
    <TouchableWithoutFeedback
      onPress={() => navigation.navigate("Details", { selectedAnimal: animal })}
    >
      <View style={styles.card}>
        <Image
          style={{
            width: "30%",
            height: "auto",
            borderTopLeftRadius: 15,
            borderBottomLeftRadius: 15,
          }}
          resizeMode="cover"
          source={{ uri: animal.img }}
        ></Image>
        <View style={{ padding: 15 }}>
          <Text style={{ fontSize: 16 }}>{animal.name}</Text>
          <Text>{animal.age} Anos</Text>
        </View>
      </View>
    </TouchableWithoutFeedback>
  );
};

export default Card;

const styles = StyleSheet.create({
  card: {
    flexDirection: "row",
    backgroundColor: "white",
    height: 110,
    borderRadius: 15,
  },
});
