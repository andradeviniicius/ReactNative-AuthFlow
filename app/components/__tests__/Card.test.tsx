import React from "react";
import { create, type ReactTestRendererJSON } from "react-test-renderer";
import Card from "../Card";
import { AnimalDetails } from "../../services/useSelectedAnimals";
import renderer from "react-test-renderer";

const mockedNavigate = jest.fn();

jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native");
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});

describe("src/components/LinkButton", () => {
  const fakeAnimal: AnimalDetails = {
    age: 1,
    categoryId: 1,
    id: "1",
    img: "",
    name: "",
    description: "",
    email: "",
    phone: "",
  };

  const CardComponent = <Card animal={fakeAnimal} key={1} />;
  test("renders correctly", () => {
    const Card = renderer.create(CardComponent).toJSON() as any;
    expect(Card).toBeTruthy();
    expect(Card.type).toBe("View");
    expect(Card.children[0].type).toBe("Image");
    expect(Card.children[1].type).toBe("View");
    expect(Card.children[1].children[0]).toBe("Text");
    expect(Card.children[1].children[1]).toBe("Text");
  });
});
