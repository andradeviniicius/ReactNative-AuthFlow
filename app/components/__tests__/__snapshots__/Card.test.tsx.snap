// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/LinkButton renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                {
                  "gap": 7,
                  "marginBottom": 50,
                  "width": "80%",
                }
              }
            >
              <Image
                source={1}
                style={
                  {
                    "height": 50,
                    "width": 50,
                  }
                }
              />
              <Text
                style={
                  {
                    "fontSize": 16,
                    "fontWeight": "bold",
                  }
                }
              >
                LOGIN
              </Text>
              <Text
                style={{}}
              >
                Insira seus dados pra continuar
              </Text>
            </View>
            <View
              style={
                {
                  "alignItems": "flex-start",
                  "margin": 10,
                  "width": "80%",
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#CB3BE6",
                    "fontWeight": "bold",
                  }
                }
              >
                E-MAIL
              </Text>
              <TextInput
                blurOnSubmit={false}
                onChangeText={[Function]}
                onFocus={[Function]}
                onSubmitEditing={[Function]}
                placeholder="ex: joÃ£ofut123@exemplo.com"
                returnKeyType="next"
                style={
                  {
                    "borderBottomWidth": 1,
                    "paddingVertical": 10,
                    "width": "100%",
                  }
                }
                value=""
              />
            </View>
            <View
              style={
                {
                  "alignItems": "flex-start",
                  "margin": 10,
                  "width": "80%",
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#CB3BE6",
                    "fontWeight": "bold",
                  }
                }
              >
                SENHA
              </Text>
              <TextInput
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="Precisa ter pelo menos 5 caracteres"
                secureTextEntry={true}
                style={
                  {
                    "borderBottomWidth": 1,
                    "paddingVertical": 10,
                    "width": "100%",
                  }
                }
                value=""
              />
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#CB3BE6",
                  "borderRadius": 25,
                  "height": 50,
                  "justifyContent": "center",
                  "marginVertical": 25,
                  "width": "80%",
                }
              }
            >
              <Text
                style={
                  {
                    "color": "white",
                    "fontSize": 16,
                    "fontWeight": "bold",
                  }
                }
              >
                Entrar
              </Text>
            </View>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Login"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
